core-api/src/services/meal.service.ts

```diff
@@ -0,0 +1,79 @@
+import db from "../config/database";
+import { Meal, SafeMeal } from "../entities/meal.entity";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import { ValidationError, validate } from "class-validator";
+import { PlaceService } from "./place.service";
+
+/**
+ * @namespace Services
+ * @class MealService
+ */
+export class MealService {
+    /**
+     * Creates a new meal in the database
+     * @param name
+     * @param description
+     * @param price
+     * @param place_id
+     */
+    public static async create(name: string, description: string, price: number, place_id: number): Promise<SafeMeal> {
+        const repository = db.getRepository(Meal);
+
+        const place = await PlaceService.findByIdWithoutSafety(place_id);
+
+        const newMeal = new Meal(name, description, price, place);
+
+        await this.validateFields(newMeal);
```

Comment:
This call is not validating whether the place exists or not. You can check this method in place.service, it does a similar thing.

Reply:
Done